clientv3/integration/watch_test.go

```diff
@@ -297,3 +297,22 @@ func putAndWatch(t *testing.T, wctx *watchctx, key, val string) {
 		}
 	}
 }
+
+func TestWatchInvalidFutureRevision(t *testing.T) {
+	defer testutil.AfterTest(t)
+
+	clus := integration.NewClusterV3(t, &integration.ClusterConfig{Size: 3})
+	defer clus.Terminate(t)
+
+	w := clientv3.NewWatcher(clus.RandClient())
+	defer w.Close()
+
+	rch := w.Watch(context.Background(), "foo", clientv3.WithRev(100))
+	wresp, ok := <-rch
+	if ok != false {
+		t.Fatalf("expected wresp 'closed'(ok false), but got ok %v", ok)
```

Comment:
@heyitsanthony So we close the channel

```go
close(ret)
pendingReq.retc <- ret
```

But we don't close `pendingReq.retc`. So it should still be open? Could you check? I might be misunderstanding.
Test fails with `ok != false`. The channel is still open.

Thanks!